gluoncv/torch/nn/nms.py

```diff
@@ -0,0 +1,159 @@
+"""Modified NMS ops as static/dynamic layers"""
+import numpy as np
+import torch
+from torchvision.ops import boxes as box_ops
+from torchvision.ops import nms  # BC-compat
+
+
+__all__ = ['batched_nms', 'ml_nms']
+
+
+def batched_nms(boxes: torch.Tensor, scores: torch.Tensor, idxs: torch.Tensor, iou_threshold: float):
+    """
+    Same as torchvision.ops.boxes.batched_nms, but safer.
+    """
+    assert boxes.shape[-1] == 4
+    # TODO may need better strategy.
+    # Investigate after having a fully-cuda NMS op.
+    if len(boxes) < 40000:
+        return box_ops.batched_nms(boxes, scores, idxs, iou_threshold)
+
+    result_mask = scores.new_zeros(scores.size(), dtype=torch.bool)
+    for iid in torch.unique(idxs).cpu().tolist():
+        mask = (idxs == iid).nonzero().view(-1)
+        keep = nms(boxes[mask], scores[mask], iou_threshold)
+        result_mask[mask[keep]] = True
+    keep = result_mask.nonzero().view(-1)
+    keep = keep[scores[keep].argsort(descending=True)]
+    return keep
+
+
+def ml_nms(boxlist, nms_thresh, max_proposals=-1, fixed_size=False):
+    """
+    Performs non-maximum suppression on a boxlist, with scores specified
+    in a boxlist field via score_field.
+
+    Args:
+        boxlist (data.structures.Boxes):
+        nms_thresh (float): the score threshold for nms
+        max_proposals (int): if > 0, then only the top max_proposals are kept
+            after non-maximum suppression
+        fixed_size (bool): force output to be static shape
+    """
+    if nms_thresh <= 0:
+        return boxlist
+    boxes = boxlist.pred_boxes.tensor
+    scores = boxlist.scores
+    labels = boxlist.pred_classes
+    keep = batched_nms(boxes, scores, labels, nms_thresh)
+    # keep = nms(boxes, scores, nms_thresh)
+    if max_proposals > 0:
+        keep = keep[: max_proposals]
+    if fixed_size:
+        boxlist._keep = keep
+        # scores = -scores
```

Comment:
Multiple commented code blocks in this file. No need to do anything if you feel necessary.

Reply:
Added comments and keep these alternative implementation in case we need that later